File: test_cron_automation.js
/**
 * Test script to verify Hugging Face Space cron job functionality.
 * This script tests the TMS Tracking API endpoints to ensure cron jobs are working.
 */

import https from 'https';
import { URL } from 'url';

const BASE_URL = 'ameetspeaks-tms.hf.space';
const VERCEL_URL = 'tms-navy-one.vercel.app';
const CRON_SECRET = 'AIC0E35w_6wXDYIz0nVtZHN59z5fUZp4o7c0bz8lz5A';

function makeRequest(hostname, path, method = 'GET', headers = {}) {
  return new Promise((resolve, reject) => {
    const options = {
      hostname: hostname,
      path: path,
      method: method,
      headers: headers
    };

    const req = https.request(options, (res) => {
      let data = '';
      res.on('data', (chunk) => { data += chunk; });
      res.on('end', () => {
        resolve({ statusCode: res.statusCode, headers: res.headers, data: data });
      });
    });

    req.on('error', (err) => { reject(err); });
    req.end();
  });
}

async function testCronAutomation() {
  console.log('🚀 Testing Hugging Face Space Cron Automation...\n');

  try {
    // Test 1: Check Hugging Face Space health
    console.log('1️⃣ Testing Hugging Face Space health...');
    const healthResponse = await makeRequest(BASE_URL, '/health');
    console.log(`   ✅ Health check: ${healthResponse.statusCode}`);
    if (healthResponse.statusCode === 200) {
      const healthData = JSON.parse(healthResponse.data);
      console.log(`   📊 Space status: ${healthData.status || 'unknown'}`);
      if (healthData.cron_jobs) {
        console.log(`   ⏰ Cron jobs: ${JSON.stringify(healthData.cron_jobs)}`);
      }
    }
    console.log('');

    // Test 2: Check cron status endpoint
    console.log('2️⃣ Testing cron status endpoint...');
    const cronStatusResponse = await makeRequest(BASE_URL, '/cron/status');
    console.log(`   ✅ Cron status: ${cronStatusResponse.statusCode}`);
    if (cronStatusResponse.statusCode === 200) {
      const cronData = JSON.parse(cronStatusResponse.data);
      console.log(`   📈 Last execution: ${cronData.last_execution || 'unknown'}`);
      console.log(`   🔢 Total executions: ${cronData.total_executions || 0}`);
    }
    console.log('');

    // Test 3: Trigger manual location poll
    console.log('3️⃣ Triggering manual location poll...');
    const locationPollResponse = await makeRequest(BASE_URL, '/location-poll', 'POST', {
      'Authorization': `Bearer ${CRON_SECRET}`,
      'Content-Type': 'application/json'
    });
    console.log(`   ✅ Location poll: ${locationPollResponse.statusCode}`);
    if (locationPollResponse.statusCode === 200) {
      const pollData = JSON.parse(locationPollResponse.data);
      console.log(`   📍 Processed: ${pollData.processed || 0} locations`);
      console.log(`   ⚠️  Errors: ${pollData.errors || 0}`);
    }
    console.log('');

    // Test 4: Trigger manual consent poll
    console.log('4️⃣ Triggering manual consent poll...');
    const consentPollResponse = await makeRequest(BASE_URL, '/consent-poll', 'POST', {
      'Authorization': `Bearer ${CRON_SECRET}`,
      'Content-Type': 'application/json'
    });
    console.log(`   ✅ Consent poll: ${consentPollResponse.statusCode}`);
    if (consentPollResponse.statusCode === 200) {
      const consentData = JSON.parse(consentPollResponse.data);
      console.log(`   📋 Processed: ${consentData.processed || 0} consents`);
      console.log(`   ⚠️  Errors: ${consentData.errors || 0}`);
    }
    console.log('');

    // Test 5: Check Vercel API endpoints
    console.log('5️⃣ Testing Vercel API endpoints...');
    
    // Test Vercel cron health
    const vercelHealthResponse = await makeRequest(VERCEL_URL, '/api/cron/health');
    console.log(`   ✅ Vercel health: ${vercelHealthResponse.statusCode}`);
    
    // Test Vercel cron monitor
    const vercelMonitorResponse = await makeRequest(VERCEL_URL, '/api/cron/monitor');
    console.log(`   ✅ Vercel monitor: ${vercelMonitorResponse.statusCode}`);
    if (vercelMonitorResponse.statusCode === 200) {
      const monitorData = JSON.parse(vercelMonitorResponse.data);
      console.log(`   📊 Monitor status: ${monitorData.status || 'unknown'}`);
      if (monitorData.monitor) {
        console.log(`   🔍 Failed jobs (24h): ${monitorData.monitor.failed_jobs_24h || 0}`);
        console.log(`   📍 Tracking health: ${monitorData.monitor.tracking_health || 'unknown'}`);
      }
    }
    console.log('');

    console.log('🎉 All tests completed successfully!');
    console.log('📋 Summary:');
    console.log('   ✅ Hugging Face Space is running');
    console.log('   ✅ Cron jobs are configured');
    console.log('   ✅ Location polling works');
    console.log('   ✅ Consent polling works');
    console.log('   ✅ Vercel API endpoints are accessible');
    
  } catch (error) {
    console.error('❌ Test failed:', error.message);
    process.exit(1);
  }
}

// Run the tests
testCronAutomation();

File: test_cron_automation.py
#!/usr/bin/env python3
"""
Test script to verify Hugging Face Space cron job functionality.
This script tests the TMS Tracking API endpoints to ensure cron jobs are working.
"""

import requests
import json
import time
from datetime import datetime

def test_space_endpoints():
    """Test the Hugging Face Space endpoints."""
    base_url = "https://ameetspeaks-tms.hf.space"
    cron_secret = "AIC0E35w_6wXDYIz0nVtZHN59z5fUZp4o7c0bz8lz5A"
    
    print(f"🚀 Testing TMS Tracking API - {datetime.now()}")
    print("=" * 60)
    
    # Test 1: Health Check
    print("1. Testing health endpoint...")
    try:
        response = requests.get(f"{base_url}/health", timeout=10)
        if response.status_code == 200:
            health_data = response.json()
            print(f"✅ Health check passed")
            print(f"   Status: {health_data.get('status')}")
            print(f"   Cron enabled: {health_data.get('cron_jobs', {}).get('enabled')}")
            print(f"   Scheduler running: {health_data.get('cron_jobs', {}).get('scheduler_running')}")
            print(f"   Active jobs: {health_data.get('cron_jobs', {}).get('jobs', [])}")
        else:
            print(f"❌ Health check failed: {response.status_code}")
            print(f"   Response: {response.text}")
    except Exception as e:
        print(f"❌ Health check error: {str(e)}")
    
    print()
    
    # Test 2: Cron Status
    print("2. Testing cron status endpoint...")
    try:
        response = requests.get(f"{base_url}/api/cron/status", timeout=10)
        if response.status_code == 200:
            cron_data = response.json()
            print(f"✅ Cron status retrieved")
            print(f"   Enabled: {cron_data.get('enabled')}")
            print(f"   Running: {cron_data.get('running')}")
            if cron_data.get('jobs'):
                for job in cron_data['jobs']:
                    print(f"   Job: {job.get('id')} - {job.get('name')}")
                    print(f"     Next run: {job.get('next_run')}")
                    print(f"     Trigger: {job.get('trigger')}")
        else:
            print(f"❌ Cron status failed: {response.status_code}")
            print(f"   Response: {response.text}")
    except Exception as e:
        print(f"❌ Cron status error: {str(e)}")
    
    print()
    
    # Test 3: Manual Location Poll Trigger
    print("3. Testing manual location poll trigger...")
    try:
        headers = {
            "Authorization": f"Bearer {cron_secret}",
            "Content-Type": "application/json"
        }
        response = requests.post(
            f"{base_url}/api/trigger/location-poll", 
            headers=headers, 
            timeout=30
        )
        if response.status_code == 200:
            result = response.json()
            print(f"✅ Location poll triggered successfully")
            print(f"   Status: {result.get('status')}")
            print(f"   Message: {result.get('message')}")
            print(f"   Timestamp: {result.get('timestamp')}")
        else:
            print(f"❌ Location poll trigger failed: {response.status_code}")
            print(f"   Response: {response.text}")
    except Exception as e:
        print(f"❌ Location poll trigger error: {str(e)}")
    
    print()
    
    # Test 4: Manual Consent Poll Trigger
    print("4. Testing manual consent poll trigger...")
    try:
        headers = {
            "Authorization": f"Bearer {cron_secret}",
            "Content-Type": "application/json"
        }
        response = requests.post(
            f"{base_url}/api/trigger/consent-poll", 
            headers=headers, 
            timeout=30
        )
        if response.status_code == 200:
            result = response.json()
            print(f"✅ Consent poll triggered successfully")
            print(f"   Status: {result.get('status')}")
            print(f"   Message: {result.get('message')}")
            print(f"   Timestamp: {result.get('timestamp')}")
        else:
            print(f"❌ Consent poll trigger failed: {response.status_code}")
            print(f"   Response: {response.text}")
    except Exception as e:
        print(f"❌ Consent poll trigger error: {str(e)}")
    
    print()
    print("=" * 60)
    print("✅ Testing completed!")
    print("\nIf all tests passed, your cron job automation is working correctly.")
    print("The system will automatically:")
    print("- Poll location data every minute")
    print("- Poll consent data every hour")
    print("- Call your Vercel API endpoints")
    print("- Keep your tracking data synchronized")

def test_vercel_endpoints():
    """Test the Vercel API endpoints that the cron jobs call."""
    vercel_url = "https://tms-navy-one.vercel.app"
    cron_secret = "AIC0E35w_6wXDYIz0nVtZHN59z5fUZp4o7c0bz8lz5A"
    
    print("\n🔍 Testing Vercel API Endpoints")
    print("=" * 40)
    
    # Test location poll endpoint
    print("Testing Vercel location poll endpoint...")
    try:
        headers = {
            "Authorization": f"Bearer {cron_secret}",
            "Content-Type": "application/json"
        }
        response = requests.post(
            f"{vercel_url}/api/cron/location-poll", 
            headers=headers, 
            timeout=30
        )
        print(f"Status: {response.status_code}")
        if response.status_code == 200:
            result = response.json()
            print(f"✅ Vercel location poll working")
            print(f"   Message: {result.get('message')}")
            print(f"   Processed: {result.get('processed')}")
            print(f"   Errors: {result.get('errors')}")
        else:
            print(f"❌ Vercel location poll failed")
            print(f"   Response: {response.text}")
    except Exception as e:
        print(f"❌ Vercel location poll error: {str(e)}")

if __name__ == "__main__":
    test_space_endpoints()
    test_vercel_endpoints()